libs/platform/form/form-generator/base-dynamic-form-generator-control.ts

```diff
@@ -65,4 +65,13 @@ export abstract class BaseDynamicFormGeneratorControl<T extends BaseDynamicFormF
         }
         return (this.formItem?.useMessageAsPlaceholder && this.formItem?.message) || '';
     }
+
+    /** @hidden */
+    get readonly(): boolean {
```

Comment:
you can simplify it:

`
get readonly(): boolean {
    return this.formItem?.readonly ?? false;
}
`